nodetool.test_drain is flaky: the mock server returns something cannot be decoded as JSON

```
failed on setup with "requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)"
...

    @pytest.fixture(scope="session")
    def rest_api_mock_server():
        ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
        port = random.randint(10000, 65535)
 ...
        while True:
            try:
>               rest_api_mock.get_expected_requests(server)

...
test/nodetool/rest_api_mock.py:236: in get_expected_requests
    return [_make_expected_request(r) for r in r.json()]

...
    def json(self, **kwargs):
        r"""Returns the json-encoded content of a response, if any.
...
>           raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
E           requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```
see https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5388/testReport/junit/test.nodetool/test_drain/Tests___Unit_Tests___test_drain/

spotted again,  see https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/5754/testReport/junit/test.nodetool/test_snapshot/Tests___Unit_Tests___test_clearnapshot/

```
failed on setup with "requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)"
...
self = <Response [200]>, kwargs = {}

    def json(self, **kwargs):
...
self = <json.decoder.JSONDecoder object at 0x7f2ffc63d650>
s = '<!doctype html><html lang="en"><head><meta charset="utf-8"/><base href="/"/><meta content="width=device-width,initial..."./images/background-wave-orig2.svg"/></div><div id="loader-block"><img src="./Loader.svg"/></div></div></body></html>'
idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/usr/lib64/python3.11/json/decoder.py:355: JSONDecodeError

During handling of the above exception, another exception occurred:

    @pytest.fixture(scope="session")
    def rest_api_mock_server():
        ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
        port = random.randint(10000, 65535)
    
        server_process = subprocess.Popen([
            sys.executable,
            os.path.join(os.path.dirname(__file__), "rest_api_mock.py"),
            ip,
            str(port)])
    
        server = (ip, port)
    
        i = 0
        while True:
            try:
>               rest_api_mock.get_expected_requests(server)

test/nodetool/conftest.py:48: 
...
self = <Response [200]>, kwargs = {}

    def json(self, **kwargs):
    
        try:
            return complexjson.loads(self.text, **kwargs)
        except JSONDecodeError as e:
            # Catch JSON-related errors and raise as requests.JSONDecodeError
            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError
>           raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
E           requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

--- a/test.py
+++ b/test.py
@@ -1015,12 +1015,15 @@ class ToolTest(Test):
 
     def __init__(self, test_no: int, shortname: str, suite) -> None:
         super().__init__(test_no, shortname, suite)
-        self.path = "pytest"
+        launcher = self.suite.cfg.get("launcher", "pytest")
+        self.path = launcher.split(maxsplit=1)[0]
         self.xmlout = os.path.join(self.suite.options.tmpdir, self.mode, "xml", self.uname + ".xunit.xml")
         ToolTest._reset(self)
 
     def _prepare_pytest_params(self, options: argparse.Namespace):
-        self.args = [
+        launcher = self.suite.cfg.get("launcher", "pytest")
+        self.args = launcher.split()[1:]
+        self.args += [
             "-s",  # don't capture print() output inside pytest
             "--log-level=DEBUG",   # Capture logs
             "-o",

--- a/test/nodetool/conftest.py
+++ b/test/nodetool/conftest.py
@@ -27,31 +27,51 @@ def pytest_addoption(parser):
                      " with --nodetool=cassandra, this should be the nodetool binary")
     parser.addoption('--jmx-path', action='store', default=None,
                      help="Path to the jmx binary, only used with --nodetool=cassandra")
+    parser.addoption('--run-within-unshare', action='store_true',
+                     help="Setup the 'lo' network if launched with unshare(1)")
 
 
 @pytest.fixture(scope="session")
-def rest_api_mock_server():
-    ip = f"127.{random.randint(0, 255)}.{random.randint(0, 255)}.{random.randint(0, 255)}"
-    port = random.randint(10000, 65535)
+def maybe_setup_loopback_network(request):
+    # unshare(1) -rn drops us in a new network namespace in which the "lo" is
+    # not up yet, so let's set it up first.
+    if request.config.getoption('--run-within-unshare'):
+        try:
+            args = "ip link set lo up".split()
+            subprocess.run(args, check=True)
+        except FileNotFoundError:
+            args = "/sbin/ifconfig lo up".split()
+            subprocess.run(args, check=True)
+    yield
 
-    server_process = subprocess.Popen([
-        sys.executable,
-        os.path.join(os.path.dirname(__file__), "rest_api_mock.py"),
-        ip,
-        str(port)])
 
+@pytest.fixture(scope="session")
+def rest_api_mock_server(request, maybe_setup_loopback_network):
+    ip = '127.0.0.1'
+    port = 12345
     server = (ip, port)
 
-    i = 0
-    while True:
+    server_process = subprocess.Popen([sys.executable,
+                                       os.path.join(os.path.dirname(__file__), "rest_api_mock.py"),
+                                       ip,
+                                       str(port)])
+    # wait 5 seconds for the expected requests
+    timeout = 5
+    interval = 0.1
+    for _ in range(int(timeout / interval)):
+        returncode = server_process.poll()
+        if returncode is not None:
+            # process terminated
+            raise subprocess.CalledProcessError(returncode, server_process.args)
         try:
             rest_api_mock.get_expected_requests(server)
             break
         except requests.exceptions.ConnectionError:
-            if i == 50:  # 5 seconds
-                raise
-            time.sleep(0.1)
-            i += 1
+            time.sleep(interval)
+    else:
+        server_process.terminate()
+        server_process.wait()
+        raise subprocess.TimeoutExpired(server_process.args, timeout)
 
     try:
         yield server

--- a/test/nodetool/suite.yaml
+++ b/test/nodetool/suite.yaml
@@ -1,2 +1,3 @@
 type: Tool
 coverage: false
+launcher: unshare -rn pytest --run-within-unshare

